### ISSUE TITLE
Build failures occurring when no file exists for the primary sim

### ISSUE THREAD

**Issue by jbphet** [phetsims/perennial#83]
There were two build failures that occurred today, both for the Swedish locale (sv), one for atomic-interactions and one for concentration.  After some investigation on this, I have a theory: This is a bit of an unusual case - atomic-interactions and concentration both have only one string in their string files - their title.  That's because the bulk of each of these simulation's strings are in some other sim - states-of-matter for atomic-interactions, beers-law-lab for concentration.  In both of these cases, the Swedish translator didn't translate that one string, so there is no string file for atomic-interactions_sv nor for concentration_sv.  This is probably a case that was never tested, i.e. the translator submits all of the shared strings but nothing for the primary sim.  We should investigate the code path in the build server where strings exist for common/shared code but there is nothing for the specified locale for the primary sim.

**Comment by jbphet** [comment]
Assigning to @mattpen for initial investigation, he can bring in @jonathanolson and/or me as needed.

**Comment by mattpen** [comment]
This failed during the `grunt production` step, and isn't really related to the build-server.  @jonathanolson can you please take a look?  The failures occured during the after-requirejs build step.

```
Running "after-requirejs-build" task
Fatal error: Cannot read property 'ATOMIC_INTERACTIONS/atomic-interactions.title' of undefined 
```

```
Running "after-requirejs-build" task
Fatal error: Cannot read property 'CONCENTRATION/concentration.title' of undefined
```

**Comment by jbphet** [comment]
Another thought: this may be a problem for any sim where the title is not translated.

**Comment by jbphet** [comment]
I'm going to add @oliver-phet as an assignee so that he can tell the Swedish translator to please resubmit atomic-interactions and concentration with the titles translated.  The title is the first item that appears on the translation page.

@oliver-phet - let me know if you need any more details on the translator.  Once he/she has been notified, you can unassign yourself.

**Comment by oliver-phet** [comment]
Translator notified.

**Comment by jbphet** [comment]
We got another one of these failures today, Feb 28 2018 at 10:01 am (approximately).  It was for phet-scale-basics, locale = mr (Marathi).  It appears to be exactly the same pattern, i.e. translators are submitting translations that exclude the title.  To help illustrate the problem, this is what you see right now if you go to the translation page, select ph-scale-basics for the sim and Marathi for the language:

![image](https://user-images.githubusercontent.com/3082831/36807078-30b137c0-1c7f-11e8-84dc-45aaadf5ee23.png)

It's odd that we are suddenly getting a rash of these, but we are, so I'm going to elevate the priority.

**Comment by jbphet** [comment]
We received two more of these this week, both were attempts to build a Maori (locale=mi) version of Circuit Construction Kit: DC - Virtual Lab.  Again, the translator didn't fill in the title, and the build failed.  Here's the build error message:

```
Build failure: Build aborted,Error: grunt build-for-server --brand=phet --locales=mi --allHTML in ../circuit-construction-kit-dc-virtual-lab failed with exit code 3 and stdout:
Running "lint-all" task

Running "clean" task

Running "requirejs-build" task

Running "after-requirejs-build" task
Fatal error: Cannot read property 'CIRCUIT_CONSTRUCTION_KIT_DC_VIRTUAL_LAB/circuit-construction-kit-dc-virtual-lab.title' of undefined  Sim = circuit-construction-kit-dc-virtual-lab Version = 1.0.2 Locales = mi Shas = ...(SHAs removed for brevity)...
```

Here's what the CCK DC - Virtual Lab page for the Maori translation:

![image](https://user-images.githubusercontent.com/3082831/41184358-09620aca-6b3d-11e8-96ec-bfd8a3539969.png)

@oliver-phet - can you please contact the Maori translator and let him know that the sim will get published if he adds a title and submits?

**Comment by oliver-phet** [comment]
Translator notified.

**Comment by jbphet** [comment]
Two more of these failures occurred this week, one for a Swedish translation of Concentration, and another for a Georgian translation of the same sim.  The Swedish translator appears to have figured it out because there is a Swedish version on the web site.  There is not, however, a Georgian version, so I think this translator should be notified.

@oliver-phet - can you let the Georgian translator know what's up?  Thanks.  Please unassign yourself once you've done this.

**Comment by oliver-phet** [comment]
Translator notified.

**Comment by jonathanolson** [comment]
It looks like this happened again. The above commit should fix it for sims published from master. @ariel-phet should this patch/change be maintenance-released?

**Comment by jbphet** [comment]
Yes, it did happen again.  @oliver-phet - can you let the Turkish translator (locale = tr), know that their translation of Area Model Decimals failed because they need to translate the title?  I can get you the ID of the translator if you need more info to know exactly who it was.

**Comment by oliver-phet** [comment]
Email to translator sent.

**Comment by ariel-phet** [comment]
@jonathanolson no, this is not appropriate for a maintenance release since it is a rare problem.

If in future the rate of this bug increases we will reconsider.

**Comment by ariel-phet** [comment]
As issues come up we will continue to document in this issue, but leaving as deferred for the moment

**Comment by jbphet** [comment]
We had three occurrances of this issue this week, all for the recently published diffusion sim, locales were iw, nb, and nn.

**Comment by jbphet** [comment]
I added code to prevent submissions of empty string files for https://github.com/phetsims/perennial/issues/83.  I still think that this should be fixed in the build server, but at least for now rosetta should be preventing this sort of build request from occurring.
